bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/DFDDataFlowAnalysisBuilder.java

```diff
@@ -76,29 +79,41 @@ public DFDDataFlowAnalysisBuilder useDataFlowDiagram(String dataFlowDiagramPath)
      * Registers a custom resource provider for the analysis
      * @param resourceProvider Custom resource provider of the analysis
      */
-    public void useCustomResourceProvider(DFDResourceProvider resourceProvider) {
+    public DFDDataFlowAnalysisBuilder useCustomResourceProvider(DFDResourceProvider resourceProvider) {
         this.customResourceProvider = Optional.of(resourceProvider);
+        return this;
```

Comment:
Is the custom resource provider set here? If not, please set customResourceProvider to the passed one